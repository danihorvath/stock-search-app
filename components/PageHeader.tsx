import { SearchResult } from "@/types/Search";
import React from "react";
import Link from "next/link";

interface PageHeaderProps {
  title?: string;
  subtitle?: string;
  children?: React.ReactNode;
  goBack?: string;
}

const PageHeader = ({ title, subtitle, children, goBack }: PageHeaderProps) => {
  return (
    <div className="flex flex-col items-center justify-center p-16 mb-12 bg-gray-800">
      {!!goBack && (
        <Link href={"/"} className="self-start mb-8">
          <button className="text-white bg-blue-500 hover:bg-blue-700  font-bold py-2 px-4 rounded">
            Go Back
          </button>
        </Link>
      )}

      <div className="text-center mb-8 text-white">
        {!!title && <h1 className="text-4xl font-bold ">{title}</h1>}
        {!!subtitle && <h2 className="text-2xl ">{subtitle}</h2>}
      </div>

      {children}
    </div>
  );
};

export default PageHeader;
