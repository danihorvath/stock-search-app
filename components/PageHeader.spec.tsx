import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import PageHeader from "./PageHeader";

const mock = {
  title: "Page Title",
  subtitle: "Page Subtitle",
  children: <div>Children</div>,
  goBack: "/",
};

test("Snapshot Test", () => {
  const { container } = render(<PageHeader {...mock} />);
  expect(container).toMatchSnapshot();
});
