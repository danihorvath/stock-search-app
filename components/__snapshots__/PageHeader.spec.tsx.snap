// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot Test 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center py-16 px-4 mb-12 bg-gray-800"
  >
    <div
      class="container"
    >
      <a
        class="self-start mb-8"
        href="/"
      >
        <button
          class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-md"
        >
          Go Back
        </button>
      </a>
      <div
        class="text-center mb-8 text-white"
      >
        <h1
          class="text-4xl font-bold "
        >
          Page Title
        </h1>
        <h2
          class="text-2xl "
        >
          Page Subtitle
        </h2>
      </div>
      <div>
        Children
      </div>
    </div>
  </div>
</div>
`;
